docs/usage/language-constraints-upgrading.md

```diff
@@ -0,0 +1,87 @@
+
+For example, consider a Node.js project which has its `engines` set to `"node": "^18.0.0 || ^20.0.0"`.
+
+Should Renovate skip Node.js v21 because it's non-LTS?
+When Node.js v22 comes out, should Renovate add it to your engines, or wait until it starts LTS?
+When Node.js v18 is EOL, should Renovate drop it from the engines list?
+
+It's hard for Renovate to know or guess what users want, because users have strong but also different opinions on the answer to the above questions.
+
+Additionally, even if Renovate guesses right or adds advanced capabilities to allow this to be configurable, then users might sit any of these "major" upgrades for months.
+If a project delays creating or merging the update to drop Node.js v18 from engines then it won't be able to upgrade to any new versions of library dependencies which themselves already dropped support.
+
+## Strict filtering limitations
+
+Consider again the Node.js project which has its `engines` set to `"node": "^18.0.0 || ^20.0.0"`.
+
+Now also consider a library which sets its `engines` to `"node": "^18.12.0 || ^20.9.0"` because it wants to only support "LTS releases" of Node.js.
+Strictly speaking, this dependency is not compatible with the project above which has wider requirements for Node versions, and Renovate would hold back any upgrades for it.
+Should Renovate somehow reason and _assume_ that this narrower engines support is actually OK?
+What if the project _already_ used an existing version of this library "incorrectly"?
+
+A second problem is that if Renovate is unable to update the language constraints, or the user ignores or does not notice the language upgrade, then they also may not realize that many of their dependencies are out of date because Renovate is not proposing PRs.
+For example, a project may have 10 dependencies, and 8 of those have updates but all 8 require the project to update its language constraints first.
+The project administrator may think they are mostly up to date, because Renovate is not proposing PRs, but in reality 80% of their dependencies are outdated.
+
+In short, users who apply `constraintsFiltering=strict` usually understimate just how strict that will be and how many releases will be filtered out.
+
+## Transitive constraint limitations
+
+It is a common problem that a library may declare language constraints (such as the `engines` from the above examples) but then itself depend on one or more libraries which have narrower contraints (such as `"node": "^20.0.0"`).
+In such scenarios Renovate "trusts" the declaration of the library and may propose it as an update even with strict constraints filtering.
+
+For some package managers, such as npm, this incompatibility won't be detected or warned (even during lock file generation), although it may also not be a problem in your application either.
```

Comment:
```suggestion
For some package managers, like `npm`, this incompatibility will _not_ be detected or warned about (even during lock file generation), but this may not be a problem for your application.
```